text/0000-source-date-epoch.md

```diff
@@ -0,0 +1,51 @@
+# Meta
+[meta]: #meta
+- Name: Source Date Epoch
+- Start Date: 2020-01-28
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Allow platforms to define an build epoch. This would improve the UX of images builds at the expense of reproducibility.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the lifecycle sets all build dates to the unix epoch for reproducibility. This can be confusing for newcomers because it looks like their builds are "40 years old". This should at the least be documented. Going one step further, the lifecycle could respect the value of `$SOURCE_DATE_EPOCH` if set by platforms, which would allow an optional override.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Lifecycle respects `$SOURCE_DATE_EPOCH` if provided by platform by using that value as the timestamp for everything it creates. This could have a default value of zero for repro. and be set optionally to allow platforms a better UX. eg. `pack build MY_APP --build-epoch 12345` would set `SOURCE_DATE_EPOCH=12345` during build  time and all layers would have that timestamp.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Lifecycle respects `$SOURCE_DATE_EPOCH` if provided by platform by using that value as the timestamp for everything it creates. This could have a default value of zero for repro. and be set optionally to allow platforms a better UX.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+More knobs to turn, more configurability.
+
+# Alternatives
+[alternatives]: #alternatives
```

Comment:
Alternative: Document the current behavior and collect feedback to determine if the ux is a problem.

Reply:
We haven't documented it, but I've so far seen 4 questions about "why are my images being created in 1980?", FWIW. At least people are taking up our releases!

Reply:
If `pack` still creates images in 1980 by default, and a flag is needed to supply the timestamp, I don't think that mitigates the initial surprise/confusion. It provides users with a mechanism to change the behavior once they understand it and if they have a reason to care about the consequences

Reply:
I'll look into why the alternatives provide this knob: the thesis is that the default for these other platforms is `timestamp == now` vs our default which is `timestamps == 1980` (set unchanging value), so the mechanism for setting it to a reproducible value was needed.

Reply:
Reading links above: in the case of `ko` they default to a fixed timestamp ([docs](https://github.com/google/ko/tree/ca1b2a1dedfdb13b4955018631fd6566e5d1a13d#why-are-my-images-all-created-in-1970)). In bazel build it seems as if the they timestamp to *now* and the have the variable for overrides.

That still points to at least *starting* with the documentation and then moving on to a knob from there.

Reply:
@zmackie should we close this RFC for now and see if we want to re-open a new one with your new data or do you want to keep this open until you get that data?